1. Added this test (**TestPeerGovernor_AddPeer_DedupesHostnameAfterDNSFailure**) - This test checks that peergov does not add the same peer twice when DNS resolution changes over time. It starts with a peer already known as relay.example.com:3001, but stored with normalized IP 44.0.0.1:3001, which represents the case where DNS worked earlier. The expected result is that the peer list still contains only one peer.

```
akhil-mac@Akhils-MacBook-Air dingo % 
akhil-mac@Akhils-MacBook-Air dingo % go test ./peergov -run 'TestPeerGovernor_AddPeer_DedupesHostnameAfterDNSFailure'                    
--- FAIL: TestPeerGovernor_AddPeer_DedupesHostnameAfterDNSFailure (0.00s)
    peergov_test.go:1262: 
                Error Trace:    /Users/akhil-mac/repos/dingo/peergov/peergov_test.go:1262
                Error:          "[0x140000c6ea0 0x140000c7040]" should have 1 item(s), but has 2
                Test:           TestPeerGovernor_AddPeer_DedupesHostnameAfterDNSFailure
FAIL
FAIL    github.com/blinklabs-io/dingo/peergov   0.348s
FAIL
akhil-mac@Akhils-MacBook-Air dingo % 
```
Without fix, the test fails before the fix because the current AddPeer only compares normalized address. During DNS failure, the new normalized address becomes relay.example.com:3001, which does not equal the existing 44.0.0.1:3001, so old code incorrectly adds a duplicate peer.

2. Added this test(**TestPeerGovernor_DenyPeer_BlocksHostnameAfterDNSFailure**) - This test checks that a denied peer cannot bypass the deny list when DNS behavior changes. It first makes DNS resolution succeed, so relay.example.com:3001 is denied as the resolved IP 44.0.0.1:3001. Then the test makes DNS resolution fail and tries to add the same hostname again. The expected result is that the peer is not added and IsDenied("relay.example.com:3001") returns true.

```
akhil-mac@Akhils-MacBook-Air dingo % go test ./peergov -run 'TestPeerGovernor_DenyPeer_BlocksHostnameAfterDNSFailure'                    
--- FAIL: TestPeerGovernor_DenyPeer_BlocksHostnameAfterDNSFailure (0.00s)
    peergov_test.go:1285: 
                Error Trace:    /Users/akhil-mac/repos/dingo/peergov/peergov_test.go:1285
                Error:          Should be empty, but was [0x140000cc4e0]
                Test:           TestPeerGovernor_DenyPeer_BlocksHostnameAfterDNSFailure
    peergov_test.go:1286: 
                Error Trace:    /Users/akhil-mac/repos/dingo/peergov/peergov_test.go:1286
                Error:          Should be true
                Test:           TestPeerGovernor_DenyPeer_BlocksHostnameAfterDNSFailure
FAIL
FAIL    github.com/blinklabs-io/dingo/peergov   0.310s
FAIL
akhil-mac@Akhils-MacBook-Air dingo % 
```

Without fix, the test fails before the fix because currently it stores only the resolved IP in the deny list. Later, when DNS fails, AddPeer checks the hostname form relay.example.com:3001, which does not match the denied IP 44.0.0.1:3001. It incorrectly allows the denied peer to be added again.

3. TestPeerGovernor_ResolveAddress_LogsDNSFailure  - This test checks that peergov logs a warning when DNS resolution fails or not.

```
akhil-mac@Akhils-MacBook-Air dingo % go test ./peergov -run 'TestPeerGovernor_ResolveAddress_LogsDNSFailure'         
--- FAIL: TestPeerGovernor_ResolveAddress_LogsDNSFailure (0.00s)
    peergov_test.go:1304: 
                Error Trace:    /Users/akhil-mac/repos/dingo/peergov/peergov_test.go:1304
                Error:          Should be true
                Test:           TestPeerGovernor_ResolveAddress_LogsDNSFailure
                Messages:       expected DNS resolution failure to be logged
FAIL
FAIL    github.com/blinklabs-io/dingo/peergov   0.407s
FAIL
akhil-mac@Akhils-MacBook-Air dingo % 
```

--- a/peergov/connections.go
+++ b/peergov/connections.go
@@ -727,14 +727,14 @@ func (p *PeerGovernor) DenyPeer(address string, duration time.Duration) {
 	normalized := p.resolveAddress(address)
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	// if idx := p.peerIndexByAddress(address); idx != -1 && p.peers[idx] != nil {
-	// 	p.denyList[p.peers[idx].NormalizedAddress] = time.Now().Add(duration)
-	// 	p.denyList[p.normalizeAddress(p.peers[idx].Address)] = time.Now().
-	// 		Add(duration)
-	//} else {
-	p.denyList[normalized] = time.Now().Add(duration)
-	// 	p.denyList[p.normalizeAddress(address)] = time.Now().Add(duration)
-	// }
+	if idx := p.peerIndexByAddress(address); idx != -1 && p.peers[idx] != nil {
+		p.denyList[p.peers[idx].NormalizedAddress] = time.Now().Add(duration)
+		p.denyList[p.normalizeAddress(p.peers[idx].Address)] = time.Now().
+			Add(duration)
+	} else {
+		p.denyList[normalized] = time.Now().Add(duration)
+		p.denyList[p.normalizeAddress(address)] = time.Now().Add(duration)
+	}
 	p.config.Logger.Debug(
 		"peer added to deny list",
 		"address", address,
@@ -748,18 +748,17 @@ func (p *PeerGovernor) DenyPeer(address string, duration time.Duration) {
 func (p *PeerGovernor) IsDenied(address string) bool {
 	// Resolve address before acquiring lock to avoid blocking DNS
 	normalized := p.resolveAddress(address)
-	// hostnameNormalized := p.normalizeAddress(address)
+	hostnameNormalized := p.normalizeAddress(address)
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	return p.isDeniedLocked(normalized)
-	// if p.isDeniedLocked(normalized) || p.isDeniedLocked(hostnameNormalized) {
-	// 	return true
-	// }
-	// if idx := p.peerIndexByAddress(address); idx != -1 && p.peers[idx] != nil {
-	// 	return p.isDeniedLocked(p.peers[idx].NormalizedAddress) ||
-	// 		p.isDeniedLocked(p.normalizeAddress(p.peers[idx].Address))
-	// }
-	// return false
+	if p.isDeniedLocked(normalized) || p.isDeniedLocked(hostnameNormalized) {
+		return true
+	}
+	if idx := p.peerIndexByAddress(address); idx != -1 && p.peers[idx] != nil {
+		return p.isDeniedLocked(p.peers[idx].NormalizedAddress) ||
+			p.isDeniedLocked(p.normalizeAddress(p.peers[idx].Address))
+	}
+	return false
 }
 
 // isDeniedLocked checks if a peer is on the deny list.

--- a/peergov/ledger_discovery.go
+++ b/peergov/ledger_discovery.go
@@ -212,8 +212,7 @@ func (p *PeerGovernor) addLedgerPeer(address string) bool {
 	p.mu.Lock()
 
 	// Check deny list
-	//if p.isDeniedLocked(normalized) || p.isDeniedLocked(p.normalizeAddress(address)) {
-	if p.isDeniedLocked(normalized) {
+	if p.isDeniedLocked(normalized) || p.isDeniedLocked(p.normalizeAddress(address)) {
 		p.mu.Unlock()
 		return false
 	}

--- a/peergov/peers.go
+++ b/peergov/peers.go
@@ -94,7 +94,7 @@ func (p *PeerGovernor) AddPeer(
 ) error {
 	// Resolve address before acquiring lock to avoid blocking DNS
 	normalized := p.resolveAddress(address)
-	// hostnameNormalized := p.normalizeAddress(address)
+	hostnameNormalized := p.normalizeAddress(address)
 
 	// Reject non-routable IPs early — topology peers bypass this check
 	// so operators can use private addresses for local relays. Inbound
@@ -114,8 +114,7 @@ func (p *PeerGovernor) AddPeer(
 
 	p.mu.Lock()
 	// Check deny list before adding
-	// if p.isDeniedLocked(normalized) || p.isDeniedLocked(hostnameNormalized) {
-	if p.isDeniedLocked(normalized) {
+	if p.isDeniedLocked(normalized) || p.isDeniedLocked(hostnameNormalized) {
 		p.config.Logger.Debug(
 			"not adding denied peer",
 			"address", address,
@@ -125,16 +124,12 @@ func (p *PeerGovernor) AddPeer(
 	}
 	// Check if already exists (use normalized address for deduplication)
 	for _, peer := range p.peers {
-		// if peer == nil {
-		// 	continue
-		// }
-		// if peer.NormalizedAddress == normalized ||
-		// 	peer.NormalizedAddress == hostnameNormalized ||
-		// 	p.normalizeAddress(peer.Address) == hostnameNormalized {
-		// 	p.mu.Unlock()
-		// 	return nil
-		// }
-		if peer != nil && peer.NormalizedAddress == normalized {
+		if peer == nil {
+			continue
+		}
+		if peer.NormalizedAddress == normalized ||
+			peer.NormalizedAddress == hostnameNormalized ||
+			p.normalizeAddress(peer.Address) == hostnameNormalized {
 			p.mu.Unlock()
 			return nil
 		}
@@ -247,12 +242,12 @@ func (p *PeerGovernor) resolveAddress(address string) string {
 	// It's a hostname - try to resolve it
 	ips, err := lookupIP(host)
 	if err != nil || len(ips) == 0 {
-		// p.config.Logger.Warn(
-		// 	"failed to resolve peer hostname",
-		// 	"address", address,
-		// 	"host", host,
-		// 	"error", err,
-		// )
+		p.config.Logger.Warn(
+			"failed to resolve peer hostname",
+			"address", address,
+			"host", host,
+			"error", err,
+		)
 		// Can't resolve, just lowercase the hostname
 		return net.JoinHostPort(strings.ToLower(host), port)
 	}

